packages/wallet/src/Transaction/ensureValidityInterval.ts

```diff
@@ -2,7 +2,8 @@ import { Cardano } from '@cardano-sdk/core';
 
 export const ensureValidityInterval = (
   currentSlot: Cardano.Slot,
+  { slotLength }: Cardano.CompactGenesis,
   validityInterval?: Cardano.ValidityInterval
 ): Cardano.ValidityInterval =>
-  // Todo: Based this on slot duration, to equal 2hrs
-  ({ invalidHereafter: Cardano.Slot(currentSlot.valueOf() + 3600), ...validityInterval });
+  // 2hrs based on slotLength
+  ({ invalidHereafter: Cardano.Slot(currentSlot.valueOf() + 2 * 3600 * slotLength.valueOf()), ...validityInterval });
```

Comment:
`2 * 3600 * slotLength.valueOf()` could be wrapped into a calc function or `2 * 3600` into a constant for better readability.

Reply:
Separate function would be nice, as that can then be unit tested with a range of values to demonstrate it's working as expected.

Reply:
Done here: [fac5fc1](https://github.com/input-output-hk/cardano-js-sdk/pull/576/commits/fac5fc14d7491710352c3c884d538d3d0d993904)